lib/Twig/Node.php

```diff
@@ -229,10 +231,18 @@ public function getIterator()
         return new ArrayIterator($this->nodes);
     }
 
-    public function getFilename()
+    public function setFilename($filename)
     {
-        if ($this->hasAttribute('module_filename')) {
-            return $this->getAttribute('module_filename');
+        $this->filename = $filename;
+        foreach ($this->nodes as $node) {
+            if (null !== $node) {
```

Comment:
Actually, there are more than just `parent` :( I will address that in another PR.